File: bench/fprimes.js
function sum_primes(upto) {
    function is_prime(num) {
        let test = 2;
        while (test * test <= num) {
            if (num % test == 0) {
                return false;
            }
            test++;
        }
        return true;
    }
    let ret = 0;
    let at = 2;
    while (at < upto) {
        if (is_prime(at)) {
            ret++;
        }
        at++;
    }
    return ret;
}

console.log(sum_primes(1000000));

File: bench/fprimes.lua
local function sum_primes(upto)
    local function is_prime(num)
        local test = 2
        while test * test <= num do
            if num % test == 0 then
                return false
            end
            test = test + 1
        end
        return true
    end
    local ret = 0
    local at = 2
    while at < upto do
        if is_prime(at) then
            ret = ret + 1
        end
        at = at + 1
    end
    return ret
end

local res = sum_primes(1000000)
print(res)